### DISCUSSION TITLE
Why use uint256 when much smaller ints will suffice?

### DISCUSSION THREAD

**Discussion by JimAtActiv** [smartcontractkit/full-blockchain-solidity-course-py#1173] [Q&A] [answered]
Hello-

Newby question, I am following Patrick's excellent course, but I have seen a number of examples where he used uint256 variables in a place where a much smaller variable, i.e. uint 16 or 32 would more than suffice for the possible range of data.  For example, the number of different tokens types that might be supported by a dApp.

Is there any point to using smaller variable types?  I would guess that larger variables result in larger contract storage, which I THINK results in higher gas fees.  Am I correct, or are the savings negligible?

Thanks! Still learning...

Jim

**Comment by n4n0b1t3** [comment, marked as answer]
the answer will be fewer gas costs, however, I have seen some examples where people tried to create test codes and measure the consumed gas for comparison. The difference will become only substantial when considering your contract will be called millions of times.

**Reply by cromewar** [reply]
Yes, if you have a contract with recurring interactions this will be meaningful, for small contracts using uint256 or uint128 will make no notable differences
